invokeai/app/services/model_install/model_install_base.py

```diff
@@ -0,0 +1,236 @@
+import traceback
+from abc import ABC, abstractmethod
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+from pydantic import BaseModel, Field
+from pydantic.networks import AnyHttpUrl
+
+from invokeai.app.services.config import InvokeAIAppConfig
+from invokeai.app.services.events import EventServiceBase
+from invokeai.app.services.invoker import Invoker
+from invokeai.app.services.model_records import ModelRecordServiceBase
+from invokeai.backend.model_manager import AnyModelConfig
+
+
+class InstallStatus(str, Enum):
+    """State of an install job running in the background."""
+
+    WAITING = "waiting"  # waiting to be dequeued
+    RUNNING = "running"  # being processed
+    COMPLETED = "completed"  # finished running
+    ERROR = "error"  # terminated with an error message
+
+
+class UnknownInstallJobException(Exception):
+    """Raised when the status of an unknown job is requested."""
+
+
+ModelSource = Union[str, Path, AnyHttpUrl]
```

Comment:
This is now replaced by:

`ModelSource = Annotated[Union[LocalModelSource, HFModelSource, URLModelSource], Body(discriminator="type")]`

I'm using `Body()` here in order to avoid the double-Annotated error in FastAPI which we previously discussed in respect to the model record storage API. No mypy errors or test failures, so seems to be halal.

Reply:
Can you add a comment for this use of `Body()`?

Relevant links:
- https://github.com/invoke-ai/InvokeAI/pull/5039#discussion_r1389752119
- https://github.com/tiangolo/fastapi/discussions/9761
- https://github.com/tiangolo/fastapi/discussions/9287

Reply:
Done.